hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/ParameterEndToEndTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression;
+
+import com.hazelcast.sql.SqlErrorCode;
+import com.hazelcast.sql.impl.expression.math.ExpressionMath;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import static com.hazelcast.sql.SqlColumnType.BIGINT;
+import static com.hazelcast.sql.SqlColumnType.BOOLEAN;
+import static com.hazelcast.sql.SqlColumnType.DECIMAL;
+import static com.hazelcast.sql.SqlColumnType.DOUBLE;
+import static com.hazelcast.sql.SqlColumnType.REAL;
+import static com.hazelcast.sql.SqlColumnType.SMALLINT;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ParameterEndToEndTest extends ExpressionEndToEndTestBase {
+
+    @Test
+    public void testBoolean() {
+        assertRow("booleanTrue and ?", EXPR0, BOOLEAN, false, false);
```

Comment:
This test mostly tests parameters with expressions but does not test parameters in isolation. The interesting part here is what would happen when the parameter's value is converted (valid or invalid), and the behavior for `SELECT ? ...`

Reply:
fixed in 3d4560a657e7723584fa0dfb6e4b7d2c498b1cfc

Reply:
Sorry, I can't remember - do we have a ticket for more specific error messages for parameters?

Reply:
here it is https://github.com/hazelcast/hazelcast/issues/17315

Reply:
Please disregard. There is an open comment for parameter behavior. Resolving this.